addons/account/models/account_bank_statement.py

```diff
@@ -109,7 +109,7 @@ def _compute_date_index(self):
         for stmt in self:
             sorted_lines = stmt.line_ids.sorted('internal_index')
             stmt.first_line_index = sorted_lines[:1].internal_index
-            stmt.date = sorted_lines.filtered(lambda l: l.state == 'posted')[-1:].date
+            stmt.date = sorted_lines.filtered(lambda l: l.state == 'posted' and l.date)[-1:].date or stmt.date
```

Comment:
All lines must have a date. It’s a required field

Reply:
That makes sense, but has it also been the case in 15?